src/js/hooks/inboundV2/addItems/useInboundAddItemsForm.jsx

```diff
@@ -0,0 +1,560 @@
+      || !formValues.values.hasManageInventory)) {
+      confirmTransition(
+        () => saveAndTransitionToNextStep(formValues, lineItems),
+        itemsWithSameCodeFlattened,
+      );
+    } else {
+      saveAndTransitionToNextStep(formValues, lineItems);
+    }
+  };
+
+  const nextPage = async () => {
+    trigger();
+    if (!isValid) return;
+
+    const formValues = getValues();
+    const lineItems = formValues.values.lineItems
+      .filter((item) => item?.product)
+      .map((item) => ({
+        ...item,
+        expirationDate: item.expirationDate ? moment(item.expirationDate)
+          .format(DateFormat.MM_DD_YYYY) : null,
+      }));
+
+    const hasInvalidQuantity = lineItems.some((item) => !item.quantityRequested || item.quantityRequested === '0');
+
+    if (hasInvalidQuantity) {
+      await confirmSave(() => checkDuplicatesSaveAndTransitionToNextStep(formValues, lineItems));
+    } else {
+      await checkDuplicatesSaveAndTransitionToNextStep(formValues, lineItems);
+    }
+  };
+
+  const saveItems = async (lineItems) => {
+    try {
+      spinner.show();
+      await saveRequisitionItemsInCurrentStep(lineItems);
+
+      Alert.success(
+        translate('react.stockMovement.alert.saveSuccess.label', 'Changes saved successfully'),
+        { timeout: 3000 },
+      );
+    } finally {
+      spinner.hide();
+    }
+  };
+
+  const save = () => {
+    trigger();
+    if (!isValid) return;
+    const lineItems = getValues()
+      .values
+      .lineItems
+      .filter((item) => Object.keys(item).length > 0);
+    const hasInvalidQuantity = lineItems.some((item) => !item.quantityRequested || item.quantityRequested === '0');
+
+    if (hasInvalidQuantity) {
+      confirmSave(() => saveItems(lineItems));
+    } else {
+      saveItems(lineItems);
+    }
+  };
+
+  const saveAndExit = async () => {
+    const formValues = getValues();
+    const lineItems = formValues.values.lineItems.filter((item) => Object.keys(item).length > 0);
+    if (isValid) {
+      try {
+        spinner.show();
+        await saveRequisitionItemsInCurrentStep(lineItems);
+        window.location = STOCK_MOVEMENT_URL.show(queryParams.id);
+      } finally {
+        spinner.hide();
+      }
+    } else {
+      confirmAlert({
+        title: translate('react.stockMovement.confirmExit.label', 'Confirm save'),
+        message: translate(
+          'react.stockMovement.confirmExit.message',
+          'Validation errors occurred. Are you sure you want to exit and lose unsaved data?',
+        ),
+        buttons: [
+          {
+            label: translate('react.default.yes.label', 'Yes'),
+            onClick: () => {
+              window.location = STOCK_MOVEMENT_URL.show(queryParams.id);
+            },
+          },
+          {
+            label: translate('react.default.no.label', 'No'),
+          },
+        ],
+      });
+    }
+  };
```

Comment:
in cases like that, I prefer to use return at the end of the if statement and then remove the else. That's not a change request, it's up to you whether you want to change it.